controllers/validator.go

```diff
@@ -13,12 +13,10 @@ import (
 
 // ValidateDataProtectionCR function validates the DPA CR, returns true if valid, false otherwise
 // it calls other validation functions to validate the DPA CR
-// TODO: #1129 Clean up duplicate logic for validating backupstoragelocations and volumesnapshotlocations in dpa
 func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error) {
-	dpa := oadpv1alpha1.DataProtectionApplication{}
-	if err := r.Get(r.Context, r.NamespacedName, &dpa); err != nil {
-		return false, err
-	}
+
+	dpa := r.dpa
```

Comment:
here is not a reconcile function. Should it update `r.dpa` after running?

Reply:
already getting the value of r.dpa once here
https://github.com/openshift/oadp-operator/pull/1316/files/59f86fd0d339b409ca723b6cb7afac71259d26df#diff-145efda6a6200cc56f1ded4099aebf0c7dd2916adeb86584fdbe5e2d2be94c79R87

Reply:
just using the local variable here for readability